website/docs: improve the docs about our Events and logging (#15270)
goauthentik/authentik: website/docs/sys-mgmt/events/notifications.md

> ---

Reviewer: github won't let me make a suggestion to line 2. But I would update the title to:

`title: Notification Rules`

And the filename also needs to be updated to `notification_rules.md`

Author: Not sure about changing the file name... let me think on what trouble that might cause... Jens very much wants to minimize the amount of redirects we do...

:::

An authentik administrator can create notification rules based on the creation of specified events. Filtering of events is processed by the authentik Policy Engine, using a combination of both 1) a policy and 2) a notification rule.

## Workflow overview

To receive notifications about events, follow this workflow:

1. Create a notification transport (or use a default notification transport).
2. Create a policy.
3. Create a notification rule, and bind the policy to the rule.

## 1. Create a notification transport

A notification transport determines the method used to deliver notifications to users. Supported delivery methods are; local notifications displayed in the authentik UI, email and webhook. Follow these [instructions](./transports.md#create-a-transport) to create a notification transport.

## 2. Create a policy

You will need to create a policy (either the **Event Matcher** policy or a custom Expression policy) that defines which events will trigger a notification.

> ### **Event Matcher** policy

Reviewer: ```suggestion
### Event Matcher policy
```

Reviewer 2: Oh wow how did that slip through

To receive notifications about events, follow this workflow:

> 1. Create a transport (or use an existing default transport)
> 2. Create a policy
> 3. Create a notification rule, and bind the policy to the rule

Reviewer: ```suggestion
1. Create a notification transport (or use a default notification transport).
2. Create a policy.
3. Create a notification rule, and bind the policy to the rule.
```

## 1. Create a notification transport

> A transport method (email, UI, webhook) is how the notifications are delivered to a user. Follow these [instructions](./transports.md#create-a-transport) for creating a transport.

Reviewer: ```suggestion
A notification transport determines the method used to deliver notifications to users. Supported delivery methods are; local notifications displayed in the authentik UI, email and webhook. Follow these [instructions](./transports.md#create-a-transport) to create a notification transport.
```

## 2. Create a policy

You will need to create a policy (either the **Event Matcher** policy or a custom Expression policy) that defines which events will trigger a notification.

### **Event Matcher** policy

> For simple filtering you can [create and configure](../../customize/policies/working_with_policies.md) a new **Event Matcher** policy to specify exactly which events (known as _Actions_ in the policy) you want to be notified about. For example, you can chose to create a policy for every time a user deletes a model object, or whenever any user fails to successfully log in.

Reviewer: ```suggestion
For simple event matching you can [create and configure](../../customize/policies/working_with_policies.md) a **Event Matcher** policy to define which events (known as _Actions_ in the policy) will trigger a notification. For example, whenever a user deletes a model object, or whenever any user fails to successfully log in.
```

return ip_address(request.context["event"].client_ip) in ip_network('192.0.2.0/24')
```

## 3. Create a notification rule and bind it to the policy

After you've created the policies to match the events you want, create a notification rule.

1. Log in as an administrator, open the authentik Admin interface, and navigate to **Event > Notification Rules**.

2. Click **Create** to add a new notification rule, or click the **Edit** icon next to an existing rule to modify it.

3. Define the policy configurations, and click **Create** or **Update** to save the settings.

- Note that policies are executed regardless of whether a group is selected. However, notifications are only triggered when a group is selected.

> - You also have to select which [notification transport](./transports.md) should be used to send the notification. Two notification transports are created by default:

Reviewer: ```suggestion
- You must select one or more [notification transports](./transports.md) to define how notifications are sent. Two notification transports are created by default:
```

Reviewer 2: 😔

    - `default-email-transport`: Delivers notifications via email using the [global email configuration](../../install-config/install/docker-compose.mdx#email-configuration-optional-but-recommended).
    - `default-local-transport`: Delivers notifications within the authentik UI.

> 4. In the list of Notification rules, click the arrow in the row of the Notification rule to expand the details of the rule.

Reviewer: ```suggestion
4. On the **Notification rules** page, click the arrow of the Notification rule to expand its policy bindings.
```

Author: sadly it is not "next to" the name of the rule... that's why I said "in the row" but that's not perfect either....

Author: but good catch on capitalization

2. Click **Create** to add a new notification rule, or click the **Edit** icon next to an existing rule to modify it.

> 3. Define the policy configurations, and click **Create** or \*\*Update to save the settings.

Reviewer: ```suggestion
3. Define the policy configurations, and click **Create** or **Update** to save the settings.
```

> - Note that you have to select which group the generated notification should be sent to. If left empty, the rule will be disabled.

Reviewer: ```suggestion
- Note that policies are executed regardless of whether a group is selected. However, notifications are only triggered when a group is selected.
```

> - You also have to select which [transports](./transports.md) should be used to send the notification. A transport with the name "default-email-transport" is created by default. This transport will use the [global email configuration](../../install-config/install/docker-compose.mdx#email-configuration-optional-but-recommended).

Reviewer: ```suggestion
- You also have to select which [notification transport](./transports.md) should be used to send the notification. Two notification transports are created by default:
    - `default-email-transport`: Delivers notifications via email using the [global email configuration](../../install-config/install/docker-compose.mdx#email-configuration-optional-but-recommended).
    - `default-local-transport`: Delivers notifications within the authentik UI.
```

4. In the list of Notification rules, click the arrow in the row of the Notification rule to expand the details of the rule.

5. Click **Bind existing Policy/Group/User**, and in the **Create Binding** modal, select the policy that you created for this notification rule and then click **Create** to finalize the binding.

:::info
Be aware that policies are executed even when no group is selected.

> :::

Reviewer: ```suggestion
```
